lua/mini/misc.lua

```diff
@@ -409,6 +413,49 @@ MiniMisc.zoom = function(buf_id, config)
   end
 end
 
+---
+MiniMisc.setup_goto_last_cursor = function()
+  if vim.fn.has("nvim-0.7.0") then
+    local group_name = "MiniMiscGotoLastCursor"
+
+    vim.api.nvim_create_augroup(group_name, { clear = true })
+    vim.api.nvim_create_autocmd("BufWinEnter", { callback = MiniMisc.goto_last_cursor, group = group_name })
+    vim.api.nvim_create_autocmd("FileType", { callback = MiniMisc.goto_last_cursor, group = group_name })
+  else
+    vim.api.nvim_command([[augroup MiniMiscGotoLastCursor]])
+    vim.api.nvim_command([[  autocmd!]])
+    vim.api.nvim_command([[  autocmd BufWinEnter * lua require('mini.misc').goto_last_cursor()]])
+    vim.api.nvim_command([[  autocmd FileType * lua require('mini.misc').goto_last_cursor()]])
+    vim.api.nvim_command([[augroup end]])
+  end
+end
+
+MiniMisc.goto_last_cursor = function()
+  local config = H.get_config()
+
+  -- Check if the buffer should be ignored
+  if vim.tbl_contains(config.cursor_goto.ignore_buftype,
+                      vim.api.nvim_buf_get_option(0, "buftype")) then
+    return
+  end
+
+  -- Check if the filetype should be ignored
+  if vim.tbl_contains(config.cursor_goto.ignore_filetype,
+                      vim.api.nvim_buf_get_option(0, "filetype")) then
+      -- reset cursor to first line
+      vim.api.nvim_command([[normal! gg]])
+      return
+  end
+
+  -- If a line has already been specified on the command line, we are done
+  --   nvim file +num
+  if vim.fn.line(".") > 1 then
+      return
+  end
+
+  H.cursor_jump()
```

Comment:
I think the whole `H.cursor_jump()` can be replaced with `vim.cmd([[normal! g'"zvzz]])`. Not sure if all those condition are vital. Was there a reason why you added them?

Reply:
They do not center all the time, only if poosible. If you're at the last line, you don't want to center, you want it to be the last line.

Reply:
> They do not center all the time, only if possible. If you're at the last line, you don't want to center, you want it to be the last line.

This is debatable. In my mind, this functionality serves two purposes:
- Reduce friction for user to move cursor to its latest place - cursor is already there.
- Reduce friction for user to search visually for cursor - it is in the center of screen and outside of folds.

The second point is not valid under current implementation. I would get it if it would require more work/code to achieve "always centered" behavior, but it is the opposite.

Reply:
It is quite strange if you have a 50 lines window, the file has 30 lines and then you center it. Then you don't see the 5 lines on the top and have to scroll.

This is what the first avoids. Does also not center because then you would have half of a blank window.

Currently it shows you always the maximum of the file content.

Reply:
And yet, always having cursor centered reduces both user surprise (expecting cursor at center) and code for implementation/testing (quite significantly).

Reply:
Always center is the default and you can disable it now.